evalml/automl/automl_search.py

```diff
@@ -947,16 +956,15 @@ def full_rankings(self):
 
     @property
     def best_pipeline(self):
-        """Returns an untrained instance of the best pipeline and parameters found during automl search.
+        """Returns a trained instance of the best pipeline and parameters found during automl search.
```

Comment:
If `self._train_best_pipeline` is true, this is correct. But lets update this description to document what happens when its false too.